src/tools/build_runner.md

```diff
@@ -0,0 +1,111 @@
+---
+title: build_runner
+description: A tool for building, testing, and running Dart code.
+---
+
+The [build_runner][] package provides general-purpose commands for generating files,
+and for optionally serving or testing the generated files.
+Read this page for an overview of using build_runner, with links to
+where you can find more information.
+For details of using build_runner with a specific package,
+see the documentation for that package.
+
+<aside class="alert alert-info" markdown="1">
+  **If you're a web developer:**
+  You can use the [`webdev` tool][webdev] instead of directly using
+  build_runner to build and serve web apps.
+</aside>
+
+The build_runner commands work with _builders_—packages
+that use the [Dart build system][build]
+to generate output files from input files.
+For example, the [json_serializable][] and [built_value_generator][]
+packages define builders that generate Dart code.
+
+Although the Dart build system is a good alternative to
+[reflection][] (which has performance issues) and
+macros (which Dart's compilers don't support),
+it can do more than just read and write Dart code.
+For example, the [sass_builder][] package implements a builder that
+generates `.css` files from `.scss` and `.sass` files.
+
+
+## Setting up build_runner
+
+To use build_runner, add a [dev dependency][] on **build_runner**
+to your app's pubspec:
+
+```
+  dev_dependencies:
+    # ···
+    build_runner: ^1.0.0
+    build_test: ^0.10.3
+```
+
+Depending on **build_test** is optional; do it if you'll be testing your code.
+
+As usual after `pubspec.yaml` changes, run `pub get` or `pub upgrade`:
+
+```terminal
+$ pub get
+```
+
+## Using built-in commands
+
+How you use the build_runner commands depends on whether you're using
+the Dart SDK or the Flutter SDK.
+Here are examples of using the build_runner **build** command:
+
+```terminal
+$ # From a directory that contains a pubspec.yaml file:
+$ pub run build_runner build  # Dart SDK
+$ flutter packages pub run build_runner build  # Flutter SDK
+```
+
+The build_runner package includes the following commands:
+
+build
+: Performs a one-time build.
+
+serve
+: Runs a development server.
+  Instead of directly using this command,
+  you can use [`webdev serve`,][webdev serve]
+  which has better default behavior.
+
+test
+: Runs [tests.][tests]
+
+watch
+: Launches a build server that watches for edits to input files.
+  Responds to changes by performing incremental rebuilds.
+
+
+## More information
+
+If you're working on web-specific code,
+see the [webdev and build_runner page.][webdev]
+
+For details on using build_runner, see the following:
+
+- Documentation for packages that require you to use build_runner.
+  These packages generally have a dependency
+  [on build][] or [on build_runner.][]
```

Comment:
Packages that would require you to use `build_runner` would only depend on `build` or maybe `build_config` - but shouldn't ever have a non-dev dependency on `build_runner`...

Reply:
Is it important to distinguish between the two types of dependencies here?

Reply:
For non-published packages it doesn't matter but we don't want people to accidentally publish with real build_runner dependencies.

Reply:
I'm not instructing people to depend on those packages... I'm only providing a link to packages that *do* have those dependencies (as reported by pub.dartlang.org). Those links were handy for me, but maybe they're TMI. I could just delete or comment out this sentence.